components/chart/src/heatfield/chart-heatfield.ts

```diff
@@ -102,6 +97,9 @@ export class DtChartHeatfield extends _DtHeatfieldMixinBase
     DtChartHeatfieldActiveChange
   >();
 
+  /** @internal The current state of the animation. */
+  _overlayState: 'void' | 'fadeIn' = 'void';
```

Comment:
```suggestion
  _overlayAnimationState: 'void' | 'fadeIn' = 'void';
```

Maybe we can call the internal variable in a way, that is specific to the animation, as it refers to the animation state, and not the state of the component.

Reply:
Sure, I'll do that.
@lukasholzer also suggested to move the animation into a separate file and make it reusable, I'll also do that right away so that I can reuse it in the upcoming PRs for the tooltip and selection area overlays.